src/services/services.ts

```diff
@@ -3705,6 +3766,39 @@ namespace ts {
                 return entries;
             }
 
+            function getJsDocCompletionEntries(fileName: string, position: number): CompletionEntry[] {
+                let sourceFile = getValidSourceFile(fileName);
+                let hasJsDocComment = ts.hasDocComment(sourceFile, position);
+                if (!hasJsDocComment) {
+                    return undefined;
+                }
+
+                // Return a list of JsDoc tag names for completion if the current position is right after an '@' sign 
+                if (sourceFile.text.charCodeAt(position - 1) === CharacterCodes.at) {
+                    return getAllJsDocCompletionEntries();
+                }
+
+                // Also return tag names if the current position is inside a tag name, e.g. (at the location of '^') 
+                //    /** @par^ */ 
+                let tag = getJsDocTagAtPosition(sourceFile, position);
+                if (tag) {
+                    if (tag.atToken.end <= position && position <= tag.tagName.end) {
+                        return getAllJsDocCompletionEntries();
+                    }
+                }
+            }
+
+            function getAllJsDocCompletionEntries(): CompletionEntry[] {
+                return ts.map(JsDocTagNames, tagName => {

```

Comment:
consider caching the  result. these are not going to change between invocations. saves you the allocations.

see `keywordCompletions` for an example.